jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js

```diff
@@ -29,7 +29,32 @@ YUI.add('charmbrowser-component', function() {
     */
     getInitialState: function() {
       // Setting a default state object.
-      return this.generateState(this.props);
+      var state = this.generateState(this.props);
+      state.headerSticky = false;
+      return state;
+    },
+
+    componentDidMount: function() {
+      this.refs.charmbrowser.addEventListener('scroll', this._onScroll);
+    },
+
+    componentWillUnmount: function() {
+      this.refs.charmbrowser.removeEventListener('scroll', this._onScroll);
+    },
+
+    /**
+      Check the scroll position and set the header to sticky.
+
+      @method _onScroll
+      @param {Object} e The scroll event
+    */
+    _onScroll: function(e) {
+      var scrollTop = e.target.scrollTop;
+      if (scrollTop > 280 && !this.state.headerSticky) {
```

Comment:
Would it be nicer to get the y position of an header plus the height to trigger instead of a hard coded number? In case styles change it will remain working.